api/src/log.rs

```diff
@@ -1,27 +1,33 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use diem_logger::{debug, Schema};
+use diem_logger::{debug, error, Schema};
 use warp::{
     http::header,
     log::{custom, Info, Log},
 };
 
 pub fn logger() -> Log<impl Fn(Info) + Copy> {
     let func = move |info: Info| {
-        debug!(HttpRequestLog {
+        let status = info.status().as_u16();
+        let log = HttpRequestLog {
             remote_addr: info.remote_addr(),
             method: info.method().to_string(),
             path: info.path().to_string(),
-            status: info.status().as_u16(),
+            status,
             referer: info.referer(),
             user_agent: info.user_agent(),
             elapsed: info.elapsed(),
             forwarded: info
                 .request_headers()
                 .get(header::FORWARDED)
-                .and_then(|v| v.to_str().ok())
-        });
+                .and_then(|v| v.to_str().ok()),
+        };
+        if status >= 500 {
+            error!(log);
+        } else {
+            debug!(log);
+        }
```

Comment:
This is good to reduce the amount of logs based on `400` errors.

We may later want to think about adding a `sample!` around the `error` call.  During a network drop, you'll likely get a flood of roughly the same error, but it really depends on the retry rate as a result.  Usually, these kinds of drops cause floods of retry requests.